File: kuikly/library/shared/src/commonMain/kotlin/com/techskillplanet/phonics/controls/TspToast.kt
package com.techskillplanet.phonics.controls

import androidx.compose.runtime.Composable
import com.tencent.kuikly.compose.foundation.background
import com.tencent.kuikly.compose.foundation.layout.Box
import com.tencent.kuikly.compose.foundation.layout.padding
import com.tencent.kuikly.compose.foundation.shape.RoundedCornerShape
import com.tencent.kuikly.compose.material3.Text
import com.tencent.kuikly.compose.ui.Alignment
import com.tencent.kuikly.compose.ui.Modifier
import com.tencent.kuikly.compose.ui.graphics.Color
import com.tencent.kuikly.compose.ui.text.font.FontWeight
import com.tencent.kuikly.compose.ui.unit.dp
import com.techskillplanet.phonics.theme.PhonicsColors

enum class TspToastVariant { Info, Success, Warning, Error }

@Composable
fun TspToast(
    message: String,
    theme: PhonicsColors,
    modifier: Modifier = Modifier,
    variant: TspToastVariant = TspToastVariant.Info,
) {
    val fill = when (variant) {
        TspToastVariant.Success -> theme.success
        TspToastVariant.Warning -> theme.warning
        TspToastVariant.Error -> theme.danger
        TspToastVariant.Info -> theme.brandDark
    }
    val color = if (variant == TspToastVariant.Warning) Color(theme.textPrimary) else Color.White
    Box(
        modifier = modifier
            .background(Color(fill), RoundedCornerShape(999.dp))
            .padding(horizontal = 16.dp, vertical = 10.dp),
        contentAlignment = Alignment.Center,
    ) {
        Text(text = message, color = color, fontWeight = FontWeight.Bold)
    }
}

File: ios-swiftui/library/Sources/PlanetComponents/TspToast.swift
import SwiftUI

/// Toast view + helper to present with RN-like auto-dismiss (~1600ms).
public struct TspToast: View {
    public static let defaultDuration: TimeInterval = 1.6

    let message: String
    let variant: String
    let theme: StarPlanetTheme

    public init(message: String, variant: String = "info", theme: StarPlanetTheme = .sky) {
        self.message = message
        self.variant = variant
        self.theme = theme
    }

    public var body: some View {
        Text(message)
            .fontWeight(.bold)
            .foregroundColor(variant == "warning" ? theme.textPrimary : .white)
            .padding(.horizontal, 14)
            .padding(.vertical, 10)
            .background(fill)
            .clipShape(Capsule())
    }

    private var fill: Color {
        switch variant {
        case "success": return theme.success
        case "warning": return theme.warning
        case "error": return theme.danger
        default: return theme.brandDark
        }
    }
}

/// Overlay host for toast visibility + auto dismiss.
public struct TspToastHost: ViewModifier {
    @Binding var visible: Bool
    let message: String
    let variant: String
    let theme: StarPlanetTheme
    let duration: TimeInterval

    public func body(content: Content) -> some View {
        content.overlay(alignment: .bottom) {
            if visible {
                TspToast(message: message, variant: variant, theme: theme)
                    .padding(.bottom, 90)
                    .transition(.opacity)
                    .onAppear {
                        DispatchQueue.main.asyncAfter(deadline: .now() + duration) {
                            visible = false
                        }
                    }
            }
        }
    }
}

public extension View {
    func tspToast(
        visible: Binding<Bool>,
        message: String,
        variant: String = "info",
        theme: StarPlanetTheme = .sky,
        duration: TimeInterval = TspToast.defaultDuration
    ) -> some View {
        modifier(TspToastHost(visible: visible, message: message, variant: variant, theme: theme, duration: duration))
    }
}
